test/SparqlAntlrParserTest.cpp

```diff
@@ -574,10 +574,9 @@ TEST(SparqlParser, DataBlock) {
   expectDataBlockFails("?test { true }");
   expectDataBlock(R"(?foo { "baz" "bar" })",
                   m::Values({Var{"?foo"}}, {{"\"baz\""}, {"\"bar\""}}));
-  // TODO<joka921/qup42> implement.
-  expectDataBlockFails(R"(( ) { })");
-  expectDataBlockFails(R"(?foo { })");
-  expectDataBlockFails(R"(( ?foo ) { })");
+  expectDataBlock(R"(( ) { })", m::Values({}, {}));
```

Comment:
Is there an e2e check or something that verifies that 
a completely empty VALUES clause (no variables and no results) doesn't break anything somewhere else? 
(This wouldn't be your fault, but it interacts with parts of the codebase in which I don't have unlimited trust for corner cases:)

Reply:
I don't now

Reply:
On the parser level those semantics are definitely fine.
For me they suffice for the sake of this PR.